.github/workflows/release.yml

```diff
@@ -0,0 +1,25 @@
+name: Release
+on:
+  push:
+    branches: [ master ]
+  release:
+    types: [ published ]
```

Comment:
I never get it right. is this going to trigger the workflow on every push to `master` _or_ release published? Or only on released published to the `master` branch?

We don’t want to trigger the workflow on every push to `master`. We only want to trigger it when we publish a (Github) release.

Reply:
> is this going to trigger the workflow on every push to `master` or _release_ published?

yes

> Or only on released published to the `master` branch?

no

> We only want to trigger it when we publish a (Github) release.

No. You want to publish both on release _and_ also when pushed to `master`. It will publish either a regular release or a SNAPSHOT one.

Reply:
Well, whatever suits you is fine to me, I just wanted to double-check that you indeed _do_ want to publish snapshot releases (this is something I disable on my projects).

Reply:
> this is something I disable on my projects

Publishing snapshots from master is _really_ nice. Especially your users will be very happy. They won't have to nag you every time they want to test out a feature that has been merged to master recently. You won't have to make a release for them. Snapshot will be published for them automatically, always.

Reply:
Shouldn’t we add

~~~ scala
ThisBuild / dynverSonatypeSnapshots := true
~~~

to the build?

Reply:
Nope, no need
https://github.com/sbt/sbt-ci-release/blob/main/plugin/src/main/scala/com/geirsson/CiReleasePlugin.scala#L105